package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -726,6 +772,670 @@ def __setstate__(self, state):
+        self._step.resize(self._step.shape[0]+1, axis=0)
+        try:
+            self._step[i] = ts.data['step']
+        except(KeyError):
+            self._step[i] = ts.frame
+        if len(self._step) > 1 and self._step[i] < self._step[i-1]:
+            raise ValueError("The H5MD standard dictates that the step "
+                             "dataset must increase monotonically in value.")
+
+        # the dataset.resize() method should work with any chunk shape
+        self._time.resize(self._time.shape[0]+1, axis=0)
+        self._time[i] = ts.time
+
+        if 'edges' in self._traj['box']:
+            self._edges.resize(self._edges.shape[0]+1, axis=0)
+            self._edges.write_direct(ts.triclinic_dimensions,
+                                     dest_sel=np.s_[i, :])
+        # These datasets are not resized if n_frames was provided as an
+        # argument, as they were initialized with their full size.
+        if self._has['position']:
+            if self.n_frames is None:
+                self._pos.resize(self._pos.shape[0]+1, axis=0)
+            self._pos.write_direct(ts.positions, dest_sel=np.s_[i, :])
+        if self._has['velocity']:
+            if self.n_frames is None:
+                self._vel.resize(self._vel.shape[0]+1, axis=0)
+            self._vel.write_direct(ts.velocities, dest_sel=np.s_[i, :])
+        if self._has['force']:
+            if self.n_frames is None:
+                self._force.resize(self._force.shape[0]+1, axis=0)
+            self._force.write_direct(ts.forces, dest_sel=np.s_[i, :])
+
+        if self.data_keys:
+            for key in self.data_keys:
+                obs = self._obsv[f'{key}/value']
+                obs.resize(obs.shape[0]+1, axis=0)
+                obs[i] = ts.data[key]
+
+        if self.convert_units:
+            self._convert_units(i)
+
+        self._counter += 1
+
+    def _convert_units(self, i):
+        """convert values in the dataset arrays with self.units dictionary"""
+
+        # Note: simply doing convert_pos_to_native(self._pos[-1]) does not
+        # actually change the values in the dataset, so assignment required
+        if self.units['time'] is not None:
+            self._time[i] = self.convert_time_to_native(self._time[i])
+        if self.units['length'] is not None:
+            if self._has['position']:
+                self._pos[i] = self.convert_pos_to_native(self._pos[i])
+            if 'edges' in self._traj['box']:
+                self._edges[i] = self.convert_pos_to_native(self._edges[i])
+        if self._has['velocity']:
+            if self.units['velocity'] is not None:
+                self._vel[i] = self.convert_velocities_to_native(self._vel[i])
+        if self._has['force']:
+            if self.units['force'] is not None:
+                self._force[i] = self.convert_forces_to_native(self._force[i])
+
+    @property
+    def has_positions(self):
```

Comment:
any ideas why the setters aren't getting covered here?

Reply:
They are never set through the setter in this code.

Will need extra tests.